crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -294,6 +294,32 @@ impl<N: ProviderNodeTypes> BlockchainProvider2<N> {
 
         Ok(None)
     }
+
+    /// Fetches data from either in-memory state or storage by [`BlockHashOrNumber`].
+    fn get_in_memory_or_storage_by_block<S, M, R>(
+        &self,
```

Comment:
Would like if the comment referred to "persistent storage", I first interpreted this as EVM storage for some reason, even though it doesn't make sense to interpret it that way